arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -1381,26 +1400,32 @@ void RocksDBEngine::processCompactions() {
           << "scheduling compaction for execution";
     
     bool queued = scheduler->queue(arangodb::RequestLane::CLIENT_SLOW, [this, bounds]() {
-      LOG_TOPIC("9485b", TRACE, Logger::ENGINES) 
-            << "executing compaction for range " << bounds;
-    
-      double start = TRI_microtime();
-      try {
-        rocksdb::CompactRangeOptions opts;
-        rocksdb::Slice b = bounds.start(), e = bounds.end();
-        _db->CompactRange(opts, bounds.columnFamily(), &b, &e);
-      } catch (std::exception const& ex) {
-        LOG_TOPIC("a4c42", WARN, Logger::ENGINES) 
-              << "compaction for range " << bounds 
-              << " failed with error: " << ex.what();
-      } catch (...) {
-        // whatever happens, we need to count down _runningCompactions in all cases
-      }
+      if (server().isStopping()) {
```

Comment:
`server()` is a method of `RocksDBEngine`, so that does not really solve the problem. I guess we need to capture a `shared_from_this` here.

Reply:
I think that is fine, because the RocksDBEngine ApplicationFeature now blocks in its shutdown until all running compaction jobs have finished.
So whenever we execute the callback, `this` is guaranteed to be still alive and kicking.

Reply:
You are right, my bad. I realized this a few minutes after I wrote that comment, but I was already AFK then.

Reply:
No worries, good that you mentioned the issue, because originally there _was_ a problem! :+1: